### PR TITLE
[inductor] Remove SIMDKernel.last_usage

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139585
* #139523
* #139452
* #139370
* #139365
* __->__ #139364

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/utils.py [hunk 1]
```diff
@@ -1051,6 +1051,19 @@ def __len__(self):
         return len(self.line)
 
 
+class DelayReplaceLine(DeferredLineBase):
```
Comment (eellison) [inline torch/_inductor/utils.py:1054]: nit: docstring/type annotations

**Review by eellison — APPROVED** [review]

**Review by shunting314 — APPROVED** [review]
The last usage (or load_counts equivalent in this PR) only consider the loads in a single kernel?

Imagine we have kernel1 followed by kernel2 both loading buf0. Would make the load in kernel1 use 'evict_last' be helpful for kernel2. Right now we seem to use 'evict_first' since the load of buf0 in kernel1 may be the last load inside this kernel.

**Comment by jansel** [comment]
I believe the cache hint stuff is just within single kernel.

**Comment by huydhn** [comment]
@pytorchbot revert -m 'Sorry for reverting your change but it is failing lots of internal tests in D65345157' -c ghfirst

The error is `NameError: name 'fused_nodes' is not defined`

**Pull request closed without merging**